### DISCUSSION TITLE
LED support for different RP2040 boards / possibility to deactivate LEDs

### DISCUSSION THREAD

**Discussion by rrottmann** [polhenarejos/pico-hsm#7] [General]
# Questions
* Maybe the onboard RGB LED could be abstracted so that it works for more of the pico-sdk supported boards?
* The firmware already supports custom extra commands. Maybe the LED could be configurable to be turned off if not needed?

# Reference Material
* Pimoroni Tiny 2040 (as used in your reference) uses 3-pin LED by default ([board schematic](https://cdn.shopify.com/s/files/1/0174/1800/files/Tiny2040_PIM558_schematic.pdf?v=1631525309)).
* Waveshare RP2040 (cheap sourced RP2040 board) uses Waveshare/NeoPixel RGB Controller [WS2812B](https://cdn-shop.adafruit.com/datasheets/WS2812B.pdf) ([schematic](https://www.waveshare.com/w/upload/4/4c/RP2040_Zero.pdf)) ([board schematic](https://www.waveshare.com/w/upload/4/4c/RP2040_Zero.pdf))

**Comment by polhenarejos** [comment]
You are right. Actually I am using a Tiny 2040 for the development, but note that not all boards have RGB led. So I decided to keep te LED part as much compatible as possible. Of course, it can be particularized to every board and probably it will provide more information rather than a blinking pattern for each state but it is not a high prioritary item.

BTW, I'll leave it open to a future feature.

**Comment by polhenarejos** [comment]
Finally I got a WS RP2040 Zero and led based on WS2812 controller is now supported.

**Reply by coelner** [reply]
I tried a M5Stack Atom S32lite, but the RGB led is not working. 
any hints or is the ws2812 support not implemented?

**Reply by polhenarejos** [reply]
It uses GPIO 48 by default. Probably your board uses another pin. You can change it through Pico Commissioner.

**Reply by coelner** [reply]
Ok, I did this. I should have told this in before. Can I dump the current configuration somehow?
